packages/flutter/lib/src/gestures/tap_and_drag.dart

```diff
@@ -1139,25 +1160,35 @@ sealed class BaseTapAndDragGestureRecognizer extends OneSequenceGestureRecognize
   String get debugDescription => 'tap_and_drag';
 
   void _acceptDrag(PointerEvent event) {
+    assert(_dragState == _DragState.accepted);
+
     if (!_wonArenaForPrimaryPointer) {
       return;
     }
+
     if (dragStartBehavior == DragStartBehavior.start) {
-      _initialPosition = _initialPosition + OffsetPair(global: event.delta, local: event.localDelta);
+      _initialPosition += OffsetPair(
+        global: event.delta,
+        local: event.localDelta,
+      );
+      _currentPosition = _initialPosition;
     }
     _checkDragStart(event);
-    if (event.localDelta != Offset.zero) {
-      final Matrix4? localToGlobal = event.transform != null ? Matrix4.tryInvert(event.transform!) : null;
-      final Offset correctedLocalPosition = _initialPosition.local + event.localDelta;
+    final Offset localDelta = event.localDelta;
```

Comment:
This should be `= _getDeltaForDetails(event.localDelta);`.

Reply:
It doesn't seem so. There is another calculation right below which will transform the delta. Updating this will lead to unexpected behavior such as the moving delta being shorter than expected. https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20framework_tests_impeller/16753/overview

Reply:
Thanks for catching that, let's leave it as is for now then.